torch/csrc/jit/graph_executor.h

```diff
@@ -26,11 +26,36 @@ struct GraphExecutorState {
   std::unordered_map<ArgumentSpec, ExecutionPlanState> execution_plans;
 };
 
+struct ExecutionPlan {
+  ExecutionPlan() = default;
+  ExecutionPlan(std::shared_ptr<Graph> graph)
+      : code(graph), graph(std::move(graph)) {}
+
+  void run(Stack& stack) const {
+    return InterpreterState(code).run(stack);
+  }
+
+  operator bool() const {
+    return static_cast<bool>(graph);
+  }
+
+  ExecutionPlanState getDebugState() {
+    ExecutionPlanState state;
+    state.code = &code;
+    state.graph = graph.get();
+    return state;
+  }
+
+  Code code;
+  std::shared_ptr<Graph> graph;
+};
```

Comment:
Why is this public now? Specialization should remain an internal detail, because we *will be changing it in the near future*.

Reply:
We will need to generate `Code` for a given graph in a `call` instruction when Interpreter hits one. This is just a starting point. What would you think would be a better interface for Interpreter to generate `Code` for a given method/graph?